### PR TITLE
Add config to enable Default Exponential Histogram for Prometheus Exporter

### PR DESCRIPTION
**Pull request by Abhishekkr3003**
This PR adds support for config `otel.exporter.prometheus.metrics.default.histogram.aggregation` by which we can make `BASE2_EXPONENTIAL_BUCKET_HISTOGRAM` as the default histogram aggregation for Prometheus exporter.

### CODE REVIEW

**Comment by jkwatson** [comment]
Hello, @Abhishekkr3003 . The environment variable/configuration option you're introducing here doesn't appear to be one of the official configuration options (see here for the full list of official env vars): https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md

In addition, I believe there is a moratorium currently on introducing new environment variables until the configuration specification is complete.

If we are going to introduce a new environment variable here, we would absolutely have to make it an "experimental" env var, and have the word "experimental" in the name of the config option. See here for an example: https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterBuilderUtil.java#L44

However, I would strongly recommend that you bring this up at the next Java SIG meeting to see if there is an appetite to add new configuration options like this.

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java
Comment (jack-berg) [inline exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java:51]: `PrometheusComponentProvider` is represents the prometheus implementation of file configuration. The schema for the config properties read here is defined in [opentelemetry-configuration](https://github.com/open-telemetry/opentelemetry-configuration/blob/670901762dd5cce1eecee423b8660e69f71ef4be/schema/meter_provider.json#L95-L118). 

I'm supportive of adding a PR to `opentelemetry-configuration` to add this property to file configuration, but let's hold off on this code until the corresponding schema exists.

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusMetricReaderProvider.java
Comment (jack-berg) [inline exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusMetricReaderProvider.java:43]:
```suggestion
        config.getString("otel.java.experimental.exporter.prometheus.metrics.default.histogram.aggregation");
```

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java
Comment (Abhishekkr3003) [reply exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java]: Sure, moved it to ExporterBuilderUtil.

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java
Comment (Abhishekkr3003) [reply exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java]: I've raised a PR (https://github.com/open-telemetry/opentelemetry-configuration/pull/97) regarding this.

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusMetricReaderProvider.java
Comment (Abhishekkr3003) [reply exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusMetricReaderProvider.java]: Sure, updated it to experimental property

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java
Comment (jack-berg) [reply exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java]: I think you meant to link to https://github.com/open-telemetry/opentelemetry-configuration/pull/99

Thanks! I'll take a look!

Comment (jack-berg) [reply exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java]: So there's a sequence of things that need to happen to unblock this:
1. Merge spec [PR](https://github.com/open-telemetry/opentelemetry-specification/pull/4124) to define new option for prometheus exporter, which unblocks:
2. Merge [PR](https://github.com/open-telemetry/opentelemetry-configuration/pull/99#pullrequestreview-2154044980) to define new `default_histogram_aggregation` option in `opentelemetry-configuration`
3. Cut a release of `opentelemetry-configuration`
4. Update `opentelemetry-java` new version of `opentelemetry-configuration`, and update `PrometheusComponentProvider` as you've done here.

Obviously that's a lot of steps. The fastest way to make this new config option usable is to wait on updating `PrometheusComponentProvider`, and rely on programmatic config for the time being (and maybe the env var if we accept it).

Comment (Abhishekkr3003) [reply exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/internal/PrometheusComponentProvider.java]: I agree with you. We can probably add config to `PrometheusComponentProvider`, once `open telemetry-configuration` is released. For now, I've removed the config from there.

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java
Comment (jack-berg) [inline exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java:118]: Can you add a test on the client side that verifies that exponential histograms are in fact working as expected when we override the default aggregation?

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/CollectorIntegrationTest.java
Comment (jack-berg) [inline exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/CollectorIntegrationTest.java:228]: This demonstrates that the configuration is not actually working. If this was an exponential histogram, `requestTotal.hasExponentialHistogram()` would be true, but its not and you're getting explicit bucket histograms instead.

In order to scrape exponential histograms, a client needs to set the accept header to `Accept: application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily`. This triggers the prometheus server to return the binary representation which includes the exponential histogram representation. This test is using the otel collector to scrape the prometheus metrics from the opentelemetry java SDK's prometheus exporter. The collector can be configured to request the binary representation when scraping as [described here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver#prometheus-native-histograms) (although I haven't tried it myself).

Comment (Abhishekkr3003) [reply exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/CollectorIntegrationTest.java:228]: I think it works as a native histogram in `CollectorIntegrationTest.java` since the number of buckets it creates for these data points is 1. At the same time, if this would have been an Explicit Bucket (which will happen if we remove `DefaultAggregationSelector=base2ExponentialBucketHistogram` from `PrometheusHttpServer`), then it will give 16 buckets for the same data points.

So this confirms that even for histograms which are not exponential, with `DefaultAggregationSelector` being set to `base2ExponentialBucketHistogram` , aggregation is indeed Native (or base2Exponential).

Comment (jack-berg) [reply exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/CollectorIntegrationTest.java:228]: This seems like a pretty brittle signal that exponential histograms have in fact been configured. Why don't we take a simpler approach and add a test to `PrometheusMetricReaderTest` which directly calls `PrometheusMetricReader#collect()`, and assert against the response `MetricSnapshots`?

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java
Comment (jack-berg) [inline exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java:549]: @Abhishekkr3003 I added a test which confirms that base2 exponential histograms are in fact configured by requesting the protobuf binary format from the prometheus server, parsing it to equivalent java bindings, and asserting against its string represetnation.

**Review by jack-berg — APPROVED** [review]

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/CollectorIntegrationTest.java
Comment (jack-berg) [reply exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/CollectorIntegrationTest.java]: Pushed a commit to resolve this.

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java
Comment (Abhishekkr3003) [reply exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java:549]: Thanks, @jack-berg , for continuing it, I have not been getting time to contribute here lately.

**Pull request merged**
